### PR TITLE
[otlphttpexporter] return nil from partial success handler when HTTP response body is empty

### PR DESCRIPTION
**Pull request by senojj**
**Description:** <Describe what has changed.>
Fixing a bug - When exporting using the otlphttpexporter, after receiving a successful HTTP response, when the response body's content length is 0 and the content type is specified as either "application/json" or "application/x-protobuf", an attempt will be made to unmarshal a nil value within any of the partial success response handler functions. This results in an error, and a potential resend of the original export request.

To fix this scenario, a check was added to the `tracesPartialSuccessHandler`, `metricsPartialSuccessHandler`, and `logsPartialSuccessHandler` functions for a `nil` value in the `protoBytes` argument. When `nil`, the function will return with a `nil` value, indicating the absence of any error.

**Link to tracking Issue:** #9666

### CODE REVIEW

**Review by TylerHelmuth — COMMENTED** [review]
_Technically_ the OTLP protocol specifies the response MUST be the signal-specific object, but I am totally ok with this defensive code. 

Can you add some new tests?

**Comment by senojj** [comment]
> _Technically_ the OTLP protocol specifies the response MUST be the signal-specific object, but I am totally ok with this defensive code.
> 
> Can you add some new tests?

I didn't realize that was specified in the OTLP protocol. Thank you for pointing that out. 

I will be happy to add some test cases to catch this.

File: exporter/otlphttpexporter/otlp_test.go [hunk 1]
```diff
@@ -467,22 +467,58 @@ func TestPartialResponse_missingHeaderButHasBody(t *testing.T) {
...
+	contentTypes := []struct {
+		contentType string
+	}{
+		{contentType: "application/x-protobuf"},
+		{contentType: "application/json"},
+	}
+	for _, tt := range contentTypes {
+		for _, telemetryType := range []string{"logs", "metrics", "traces"} {
+			t.Run("Missing header but has body for content type "+tt.contentType+" "+telemetryType, func(t *testing.T) {
+				response := ptraceotlp.NewExportResponse()
```
Comment (TylerHelmuth) [inline exporter/otlphttpexporter/otlp_test.go:479]: For success they probably all look the same, but technically each signal has their own expected response object.  Can you update the test to use the proper response per signal?

Comment (senojj) [reply exporter/otlphttpexporter/otlp_test.go:479]: good catch, will update.

Comment (senojj) [reply exporter/otlphttpexporter/otlp_test.go]: I _think_ I now have the tests in a good place. I apologize for so much back and forth as I am getting used to the project details.

**Review by iblancasa — APPROVED** [review]

File: exporter/otlphttpexporter/otlp_test.go [hunk 2]
```diff
@@ -524,53 +634,145 @@ func TestPartialSuccess_shortContentLengthHeader(t *testing.T) {
...
+				var data []byte
+				var err error
+
+				switch ct.contentType {
+				case jsonContentType:
+					data, err = serializer.MarshalJSON()
+				case protobufContentType:
+					data, err = serializer.MarshalProto()
+				default:
+					panic(ct.contentType)
```
Comment (TylerHelmuth) [inline exporter/otlphttpexporter/otlp_test.go:680]: Lets to `require.Fail` instead of a panic.

**Review by TylerHelmuth — APPROVED** [review]

**Comment by dmitryax** [comment]
@senojj thanks for the fix. Can you please add a changelog entry?

**Comment by senojj** [comment]
> @senojj thanks for the fix. Can you please add a changelog entry?

This is done. I apologize for the delay in adding that.

**Review by dmitryax — APPROVED** [review]

**Pull request merged**